pkg/agent/qrm-plugins/cpu/dynamicpolicy/state/util.go

```diff
@@ -490,3 +430,67 @@ func checkCPUSetMap(map1, map2 map[int]machine.CPUSet) bool {
 func CPUPreciseCeil(request float64) int {
 	return int(math.Ceil(float64(int(request*1000)) / 1000))
 }
+
+// GenerateMachineStateFromPodEntriesByPolicy returns NUMANodeMap for given resource based on
+// machine info and reserved resources along with existed pod entries and policy name
+func GenerateMachineStateFromPodEntriesByPolicy(topology *machine.CPUTopology, podEntries PodEntries, policyName string) (NUMANodeMap, error) {
```

Comment:
Why the GenerateMachineStateFromPodEntriesByPolicy is in dynamicpolicy state util, and not a  common place for both dynamicpolicy and nativepolicy

Reply:
@luomingmeng Good Question. Since the GenerateMachineStateFromPodEntries method depends on too many other methods and structures within the state package, extracting it as a standalone utility would require extracting the entire state package. However, doing so would incur a significant refactoring cos

Reply:
ok, add some todo comments for future refinement

Reply:
@luomingmeng done